tiledb/sm/serialization/capnp_utils.h

```diff
@@ -89,6 +89,20 @@ Status attribute_to_capnp(
 tuple<Status, optional<shared_ptr<Attribute>>> attribute_from_capnp(
     const capnp::Attribute::Reader& attribute_reader);
 
+inline std::string serialize_uri_to_relative(
```

Comment:
Needs comment to explain the motivation for `last_path_part` vs `last_two_path_parts` based on version.
Also, it would be nice to rename it to something like `serialize_array_dir_uri_to_relative` to get rid of confusion that function might apply to all URIs.

Reply:
Thanks for the comments! 
I followed your advice and renamed, however I think the way I refactored that function to fix the issue you pointed out (that each fragment can have different format version according to the time it was written) it can serve for other relative-absolute path conversions as well in the codebase. To be seen when we'll address https://app.shortcut.com/tiledb-inc/story/23404/serialization-relative-uri-paths-should-be-calculated-based-on-each-fragment-write-version